tests/php/lib/class-wp-test-spy-rest-server.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+class WP_Test_Spy_REST_Server extends WP_REST_Server {
+	/**
+	 * Get the raw $endpoints data from the server
+	 *
+	 * @return array
+	 */
+	public function get_raw_endpoint_data() {
+		return $this->endpoints;
+	}
+
+	/**
+	 * Allow calling protected methods from tests
+	 *
+	 * @param string $method Method to call
+	 * @param array $args Arguments to pass to the method
+	 * @return mixed
+	 */
+	public function __call( $method, $args ) {
+		return call_user_func_array( array( $this, $method ), $args );
+	}
+
+	/**
+	 * Call dispatch() with the rest_post_dispatch filter
+	 */
+	public function dispatch( $request ) {
+		$result = parent::dispatch( $request );
+		$result = rest_ensure_response( $result );
+		if ( is_wp_error( $result ) ) {
+			$result = $this->error_to_response( $result );
+		}
+		return apply_filters( 'rest_post_dispatch', rest_ensure_response( $result ), $this, $request );
```

Comment:
It's unnecessary to call `rest_ensure_response( $result )` again when it was done in line 29. It's not even necessary to do it inside the `if` because `WP_REST_Server->error_to_response` already converts it to a `WP_REST_Response`, which is what `rest_ensure_response` does.